mongoengine/base/proxy.py

```diff
@@ -186,7 +198,17 @@ def _get_current_object(self):
             collection = self.__document_type._get_collection()
             son = collection.find_one({'_id': self.__pk})
             if son is None:
-                raise DoesNotExist('Document has been deleted.')
+                error =  'Document ({} {}'.format(
+                    collection.name, self.__pk
+                )
+                if self.__parent_ref_instance:
+                    error += ', referenced by {} {} in field {}'.format(
+                        self.__parent_ref_instance.__class__.__name__,
+                        self.__parent_ref_instance.pk,
+                        self.__parent_ref_field_name,
+                    )
+                error += ') has been deleted.'
+                raise DoesNotExist(error)
```

Comment:
Can we also add the extra data as exception attributes? At some point we would want to inspect the parent object and parsing error message is not a good idea.

Reply:
If we do this, we should start with doing it in the upstream version so that we have a consistent interface. A more informative message on the same exception is good. Raising what would essentially amount to a different exception is not good.